lib/base-compiler.js

```diff
@@ -172,12 +179,27 @@ Compile.prototype.compile = function (source, options, filters) {
                         result.asm = "<Compilation failed>";
                         return result;
                     }
+                    result.hasOptOutput = false;
+                    if(self.couldHaveOptOutput(options, self.compiler.version)) {
+                        const optPath = path.join(dirPath, outputFilebase + ".opt.yaml");
```

Comment:
Sticking to the Ubuntu LTS version sound good. I'd like an excuse to move to the new goodness! 16.04 ships with:
```
node --version
v4.2.6
```
which seems to support all this stuff.